lib/rooted/adapters/rabbit_mq.rb

```
@@ -0,0 +1,44 @@
+require 'bunny'
+require 'rooted/errors'
+
+module Rooted
+  module Adapters
+    class RabbitMQ
+      def initialize(config = {})
+        @config = config || {}
+        @connection = Bunny.new(@config).start
+        @exchange = declare_exchange!
+      end
+
+      def publish(resource_class, verb, representation, options = {})
+        @exchange.publish(
+          representation,
+          routing_key: routing_key(resource_class, verb),
+          persistent: @config.fetch(:persistent, true),
+          timestamp: Time.now.to_i,
+          content_type: options[:content_type])
+        raise NotPublished unless @exchange.wait_for_confirms
```

Yep that was pretty much my take on it. I was assuming we'd want to use publisher confirms because the events are going to be reasonably key in some situations so we'd want to know if they weren't being published.

Need to have a bit of a think about how to handle this if we use the `after_commit` callback because Rails will swallow the errors (changing in Rails 5 where it will throw them by default, although it's a config setting). But I figured it's better to know about it rather than just have the events lost silently.